bin/node/src/config.rs

```diff
@@ -41,54 +77,47 @@ mod tests {
                 NODE_CONFIG_FILE_PATH,
                 r#"
                     [block_producer]
-                    endpoint = { host = "127.0.0.1",  port = 8080 }
-                    store_url = "http://store:8000"
+                    endpoint = { host = "block_host",  port = 1133 }
                     verify_tx_proofs = true
 
                     [rpc]
-                    endpoint = { host = "127.0.0.1",  port = 8080 }
-                    store_url = "http://store:8000"
-                    block_producer_url = "http://block_producer:8001"
+                    endpoint = { host = "rpc_host",  port = 4321 }
 
                     [store]
-                    endpoint = { host = "127.0.0.1",  port = 8080 }
+                    endpoint = { host = "store_host",  port = 1234 }
                     database_filepath = "local.sqlite3"
                     genesis_filepath = "genesis.dat"
                     blockstore_dir = "blocks"
                 "#,
             )?;
 
-            let config: NodeConfig =
-                load_config(PathBuf::from(NODE_CONFIG_FILE_PATH).as_path()).extract()?;
+            let config = NodeConfig::load(PathBuf::from_str(NODE_CONFIG_FILE_PATH).unwrap())?;
 
             assert_eq!(
                 config,
                 NodeConfig {
-                    block_producer: Some(BlockProducerConfig {
+                    block_producer: BlockProducerConfig {
                         endpoint: Endpoint {
-                            host: "127.0.0.1".to_string(),
-                            port: 8080,
+                            host: "block_host".to_string(),
+                            port: 1133,
                         },
-                        store_url: "http://store:8000".to_string(),
+                        store_url: "http://store_host:1234".to_string(),
                         verify_tx_proofs: true
-                    }),
-                    rpc: Some(RpcConfig {
-                        endpoint: Endpoint {
-                            host: "127.0.0.1".to_string(),
-                            port: 8080,
-                        },
-                        store_url: "http://store:8000".to_string(),
-                        block_producer_url: "http://block_producer:8001".to_string(),
-                    }),
-                    store: Some(StoreConfig {
+                    },
+                    rpc: RpcConfig {
+                        endpoint: Endpoint { host: "rpc_host".to_string(), port: 4321 },
+                        store_url: "http://store_host:1234".to_string(),
+                        block_producer_url: "http://block_host:1133".to_string(),
```

Comment:
I'm also curious - is our assumption that these are always `http`?

Reply:
hmmm - not sure. Probably, we should not assume this.